model/src/main/kotlin/io/spine/validation/required/RequiredRule.kt

```diff
@@ -34,22 +34,47 @@ import io.spine.protodata.ast.isMap
 import io.spine.protodata.value.Value
 import io.spine.validate.Diags.Required.collectionErrorMsg
 import io.spine.validate.Diags.Required.singularErrorMsg
+import io.spine.validation.ComparisonOperator
 import io.spine.validation.DefaultErrorMessage.from
+import io.spine.validation.Rule
+import io.spine.validation.SimpleRule
+import io.spine.validation.UnsetValue
+import io.spine.validation.boolValue
+import io.spine.validation.extractType
+import io.spine.validation.`is`
+import io.spine.validation.simpleRule
+import io.spine.validation.wrap
 
 /**
  * A factory of [SimpleRule]s which represent the `(required)` constraint.
  */
 internal object RequiredRule {
 
+    /**
+     * Checks if the given field is required by inspecting its options.
+     *
+     * Some fields such as entity state IDs or command message IDs are assumed to be present.
+     * For such fields pass the parameter [assumed] set to `true`.
+     * In such a case, only setting `(required) = false` for the field would cancel the assumption.
+     *
+     * @param field The field.
+     * @param assumed If set to `true`, the method returns `false` only if the field has
+     *  the `(required)` option explicitly set to `false`.
+     * @return `true` if the field is required or assumed required.
+     */
+    fun isRequired(field: Field, assumed: Boolean): Boolean =
```

Comment:
I've moved this function to the top because it's the one from which we start dealing with the option processing.